src/platform_impl/macos/menu.rs

```diff
@@ -1,117 +1,292 @@
+                        let title = NSString::alloc(nil).init_str("Services");
+                        let item = make_menu_item(title, None, None);
+                        let app_class = class!(NSApplication);
+                        let app: id = msg_send![app_class, sharedApplication];
+                        let services: id = msg_send![app, servicesMenu];
+                        let _: () = msg_send![&*item, setSubmenu: services];
+                        item
+                    }
+                };
 
-        // Seperator menu item
-        let sep_first = NSMenuItem::separatorItem(nil);
-
-        // Hide application menu item
-        let hide_item_prefix = NSString::alloc(nil).init_str("Hide ");
-        let hide_item_title = hide_item_prefix.stringByAppendingString_(process_name);
-        let hide_item = menu_item(
-            hide_item_title,
-            selector("hide:"),
-            Some(KeyEquivalent {
-                key: "h",
-                masks: None,
-            }),
-        );
-
-        // Hide other applications menu item
-        let hide_others_item_title = NSString::alloc(nil).init_str("Hide Others");
-        let hide_others_item = menu_item(
-            hide_others_item_title,
-            selector("hideOtherApplications:"),
-            Some(KeyEquivalent {
-                key: "h",
-                masks: Some(
-                    NSEventModifierFlags::NSAlternateKeyMask
-                        | NSEventModifierFlags::NSCommandKeyMask,
-                ),
-            }),
-        );
+                menu_object.addItem_(item_obj);
+            }
 
-        // Show applications menu item
-        let show_all_item_title = NSString::alloc(nil).init_str("Show All");
-        let show_all_item = menu_item(
-            show_all_item_title,
-            selector("unhideAllApplications:"),
-            None,
-        );
-
-        // Seperator menu item
-        let sep = NSMenuItem::separatorItem(nil);
-
-        // Quit application menu item
-        let quit_item_prefix = NSString::alloc(nil).init_str("Quit ");
-        let quit_item_title = quit_item_prefix.stringByAppendingString_(process_name);
-        let quit_item = menu_item(
-            quit_item_title,
-            selector("terminate:"),
-            Some(KeyEquivalent {
-                key: "q",
-                masks: None,
-            }),
-        );
+            menu_item.setSubmenu_(menu_object);
+        }
 
-        app_menu.addItem_(about_item);
-        app_menu.addItem_(sep_first);
-        app_menu.addItem_(hide_item);
-        app_menu.addItem_(hide_others_item);
-        app_menu.addItem_(show_all_item);
-        app_menu.addItem_(sep);
-        app_menu.addItem_(quit_item);
-        app_menu_item.setSubmenu_(app_menu);
+        // Set the menu as main menu for the app
+        let app = NSApp();
+        app.setMainMenu_(menubar);
     });
 }
 
-fn menu_item(
+fn make_menu_item(
```

Comment:
Consider making this function take a &str as the first parameter to and turning that into the NSString in the function. This would make the code cleaner.